pkg/operator/controllers/gameserver_controller.go

```diff
@@ -98,48 +96,19 @@ func (r *GameServerReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	var gs mpsv1alpha1.GameServer
 	if err := r.Get(ctx, req.NamespacedName, &gs); err != nil {
 		if apierrors.IsNotFound(err) {
-			log.Info("Unable to fetch GameServer - skipping")
-			return ctrl.Result{}, nil
-		}
-		log.Error(err, "unable to fetch GameServer")
-		return ctrl.Result{}, err
-	}
-
-	// ----------------------- finalizer logic start ----------------------- //
-	// examine DeletionTimestamp to determine if object is under deletion
-	if gs.ObjectMeta.DeletionTimestamp.IsZero() {
-		// The object is not being deleted, so if it does not have our finalizer,
-		// then lets add the finalizer and update the object. This is equivalent
-		// registering our finalizer.
-		if !containsString(gs.GetFinalizers(), finalizerName) {
-			patch := client.MergeFrom(gs.DeepCopy())
-			controllerutil.AddFinalizer(&gs, finalizerName)
-			if err := r.Patch(ctx, &gs, patch); err != nil {
-				return ctrl.Result{}, err
-			}
-			return ctrl.Result{}, nil
-		}
-	} else {
-		// The object is being deleted
-		if containsString(gs.GetFinalizers(), finalizerName) {
-			patch := client.MergeFrom(gs.DeepCopy())
-			// our finalizer is present, so lets handle any external dependency
-			err := r.unassignPorts(&gs)
+			log.Info("Unable to fetch GameServer, it has probably been deleted. Trying to deregister ports")
```

Comment:
it's unregister I think?

Reply:
had to look it up :D https://english.stackexchange.com/questions/25931/unregister-vs-deregister

Both seem to be "correct", so let's just keep the "deregister". Thanks!